newsroom/topics_folders/folders.py

```diff
@@ -0,0 +1,116 @@
+from enum import Enum, unique
+from typing import Optional, Annotated
+
+from newsroom import MONGO_PREFIX
+from newsroom.core.resources.model import NewshubResourceModel
+from newsroom.core.resources.service import NewshubAsyncResourceService
+
+# from newsroom.signals import user_deleted
+
+from newsroom.topics.topics_async import TopicService
+
+from superdesk.core.resources.fields import ObjectId as ObjectIdField
+from superdesk.core.resources import (
+    ResourceConfig,
+    MongoIndexOptions,
+    MongoResourceConfig,
+    RestEndpointConfig,
+    RestParentLink,
+)
+from superdesk.core.resources.validators import validate_data_relation_async
+
+# from superdesk.core.module import SuperdeskAsyncApp
+
+
+@unique
+class SectionType(str, Enum):
+    WIRE = "wire"
+    AGENDA = "agenda"
+    MONITORING = "monitoring"
+
+
+class FolderResourceModel(NewshubResourceModel):
+    name: str
+    parent: Annotated[Optional[ObjectIdField], validate_data_relation_async("topic_folders")] = None
+    section: SectionType
+
+
+class FolderResourceService(NewshubAsyncResourceService[FolderResourceModel]):
+    async def on_deleted(self, doc):
+        await self.delete({"parent": doc["_id"]})
```

Comment:
These 3 `delete` methods should use the newer [delete_many](https://github.com/superdesk/superdesk-core/blob/async/superdesk/core/resources/service.py#L412) method on the newer service.